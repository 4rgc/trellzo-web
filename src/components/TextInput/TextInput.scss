input,
textarea {
	border-radius: 5px;
	border: darkgrey 1px solid;
	padding: 5px;
	box-shadow: 1px grey;
	outline: rgba(0, 0, 0, 0) solid 0px;
	transition: outline 50ms linear;
	box-sizing: border-box;
	width: 100%;

	&.tinput-md {
		font-size: 0.85rem;
	}

	&.tinput-lg {
		font-size: large;
	}
}

textarea {
	resize: none;
}

input:focus-visible,
textarea:focus-visible {
	outline: rgba(20, 100, 255, 50) solid 2px;
}
