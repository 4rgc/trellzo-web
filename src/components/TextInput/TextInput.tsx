import { ComponentProps } from 'react';
import './TextInput.scss';

type InputChangeHandler = (text: string) => void;

interface ISingleLineInputProps {
	variant?: 'single';
	onChange?: InputChangeHandler;
	fontSize?: 'md' | 'lg';
	value?: string;
}

interface IMultilineInputProps {
	variant?: 'multi';
	onChange?: InputChangeHandler;
	fontSize?: 'md' | 'lg';
	value?: string;
	rows?: number;
	cols?: number;
}

export type TextInputProps = (ISingleLineInputProps | IMultilineInputProps) &
	Omit<ComponentProps<'input'> & ComponentProps<'textarea'>, 'onChange'>;

const TextInput: React.FC<TextInputProps> = (props) => {
	const {
		variant = 'single',
		onChange,
		value,
		fontSize = 'md',
		...otherProps
	} = props;

	const className = `tinput-${fontSize}`;

	return variant === 'multi' ? (
		<textarea
			className={className}
			onChange={onChange && ((e) => onChange(e.target.value))}
			value={value}
			{...otherProps}
		>
			{value}
		</textarea>
	) : (
		<input
			className={className}
			value={value}
			onChange={onChange && ((e) => onChange(e.target.value))}
			{...otherProps}
		/>
	);
};

export default TextInput;
