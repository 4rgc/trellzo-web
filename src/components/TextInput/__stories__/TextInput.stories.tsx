import TextInput from '../TextInput';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { useState } from 'react';

export default {
	title: 'Components/TextInput',
	component: TextInput,
	args: {
		variant: 'single',
	},
	argTypes: {
		onChanged: { action: 'clicked' },
	},
} as ComponentMeta<typeof TextInput>;

const Template: ComponentStory<typeof TextInput> = (args) => {
	const [value, setValue] = useState('');

	return (
		<div
			style={{
				padding: '20px',
				border: '2px black solid',
				width: '400px',
			}}
		>
			<TextInput {...args} value={value} onChanged={setValue} />
		</div>
	);
};

export const Default = Template.bind({});

export const WithPlaceholder = Template.bind({});
WithPlaceholder.args = { placeholder: 'Enter something...' };

export const SingleLineWithRowsSpecified = Template.bind({});
SingleLineWithRowsSpecified.args = { rows: 5 };

export const Multiline = Template.bind({});
Multiline.args = { variant: 'multi' };

export const BiggerMultiline = Template.bind({});
BiggerMultiline.args = { variant: 'multi', rows: 6 };

export const MultilineWithPlaceholder = Template.bind({});
MultilineWithPlaceholder.args = {
	variant: 'multi',
	placeholder: 'Type something long here...',
};

export const LargeFont = Template.bind({});
LargeFont.args = { fontSize: 'lg' };

export const InATightContainer: ComponentStory<typeof TextInput> = (args) => {
	const [value, setValue] = useState('');

	return (
		<div
			style={{
				border: '2px black solid',
				width: '100px',
				padding: '10px',
			}}
		>
			<TextInput {...args} value={value} onChanged={setValue} />
		</div>
	);
};
