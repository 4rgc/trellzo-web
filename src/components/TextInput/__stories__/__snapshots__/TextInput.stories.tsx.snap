// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/TextInput BiggerMultiline smoke-test 1`] = `
<div style="padding: 20px; border: 2px solid black; width: 400px;">
  <textarea class="tinput-md"
            rows="6"
  >
  </textarea>
</div>
`;

exports[`Components/TextInput Default smoke-test 1`] = `
<div style="padding: 20px; border: 2px solid black; width: 400px;">
  <input class="tinput-md"
         value
  >
</div>
`;

exports[`Components/TextInput InATightContainer smoke-test 1`] = `
<div style="border: 2px solid black; width: 100px; padding: 10px;">
  <input class="tinput-md"
         value
  >
</div>
`;

exports[`Components/TextInput LargeFont smoke-test 1`] = `
<div style="padding: 20px; border: 2px solid black; width: 400px;">
  <input class="tinput-lg"
         value
  >
</div>
`;

exports[`Components/TextInput Multiline smoke-test 1`] = `
<div style="padding: 20px; border: 2px solid black; width: 400px;">
  <textarea class="tinput-md">
  </textarea>
</div>
`;

exports[`Components/TextInput MultilineWithPlaceholder smoke-test 1`] = `
<div style="padding: 20px; border: 2px solid black; width: 400px;">
  <textarea class="tinput-md"
            placeholder="Type something long here..."
  >
  </textarea>
</div>
`;

exports[`Components/TextInput SingleLineWithRowsSpecified smoke-test 1`] = `
<div style="padding: 20px; border: 2px solid black; width: 400px;">
  <input class="tinput-md"
         rows="5"
         value
  >
</div>
`;

exports[`Components/TextInput WithPlaceholder smoke-test 1`] = `
<div style="padding: 20px; border: 2px solid black; width: 400px;">
  <input class="tinput-md"
         placeholder="Enter something..."
         value
  >
</div>
`;
